ingestion/src/metadata/data_insight/processor/web_analytic_report_data_processor.py

```diff
@@ -107,12 +108,27 @@ def _refine_entity_event(self) -> Generator[dict, WebAnalyticEventData, None]:
                 continue
 
             entity_obj = EntityObj(split_url[0], split_url[1])
+            entity_type = entity_obj.entity_type
+            re_pattern = re.compile(f"(.*{entity_type}/{entity_obj.fqn})")
+
+            if (
+                entity_obj.fqn in refined_data
+                and not refined_data[entity_obj.fqn]["entityHref"]
+            ):
+                # if we've seen the entity previously but were not able to get
+                # the URL we'll try again from the new event.
+                try:
+                    entity_href = re.search(re_pattern, event.eventData.fullUrl).group(
+                        1
+                    )
+                except IndexError:
+                    logger.warning(f"Could not find entity Href for {entity_obj.fqn}")
```

Comment:
If this error is expected and could add a lot of noise in the logs I would change the level to debug.